thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/cache/CacheDataSource.java

```diff
@@ -19,30 +19,36 @@
 
 package org.apache.pinot.thirdeye.detection.cache;
 
+import java.util.Map;
+
+
 /**
  * Config for a single centralized cache data source.
  * For example, this class could be for Couchbase, or Redis, or Cassandra, etc.
  */
 public class CacheDataSource {
 
   /**
-   * authentication stuff.
+   * class name, e.g. org.apache.pinot.thirdeye.detection.cache.CouchbaseCacheDAO
+   */
+  private String className;
+
+  /**
+   * settings/config for the specific data source. generic since different
+   * data stores may have different authentication methods.
    */
-  private String host;
-  private String authUsername;
-  private String authPassword;
-  private String bucketName;
+  private Map<String, Object> config;
 
   // left blank
   public CacheDataSource() {}
 
-  public String getHost() { return host; }
-  public String getAuthUsername() { return authUsername; }
-  public String getAuthPassword() { return authPassword; }
-  public String getBucketName() { return bucketName; }
+  public String getClassName() { return className; }
+  public Map<String, Object> getConfig() { return config; }
+
+  public Object getField(String fieldName) { return config.get(fieldName); }
+  public String getFieldAsString(String fieldName) { return String.valueOf(config.get(fieldName)); }
+  public Boolean getFieldAsBoolean(String fieldName) { return Boolean.parseBoolean(this.getFieldAsString(fieldName)); }
```

Comment:
Why not using Configuration interface which has all the helper functions built in?

Reply:
I'm not sure which Configuration interface you're referring to. I switched to using MapUtils for now.

Reply:
https://commons.apache.org/proper/commons-configuration/apidocs/org/apache/commons/configuration2/Configuration.html

Reply:
Seems like Jackson doesn't like the Configuration interface, and isn't able to deserialize into it directly. I think keeping Map<String, Object> is fine if we reuse methods from MapUtils.